lib/elixir/lib/kernel.ex

```diff
@@ -3627,7 +3627,9 @@ defmodule Kernel do
           unquote(expanded).__using__(unquote(opts))
         end
       _otherwise ->
-        raise ArgumentError, "invalid arguments for use, expected an atom or alias as argument"
+        IO.inspect module
+        IO.inspect __CALLER__
+        raise ArgumentError, "invalid arguments for use, expected a compile time atom or alias, got: #{Macro.to_string(module)}"
```

Comment:
In case of multi-alias call (`use Foo.{Bar, 42}`) it'll display whole argument for `use`, it might be better to limit it to the actually wrong part.